_posts/2017-09-08-web.md

```diff
@@ -12,9 +12,84 @@ hierarchy:
+
+##### Internet Addressing
+
+When the Internet was first founded, it nothing more than a network of computers for academic purposes; there was no way to go to `google.com`, because there was no concept of a named website, or a *domain*. Computers were connected to each other solely on the basis of an address of the form *xxx.xxx.xxx.xxx*, where *x* is a number from *0 - 255*.
+
+> More recently, these addresses can also take a form of *xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx*, where *x* is a [hexadecimal number](https://en.wikipedia.org/wiki/Hexadecimal) between 0 and 15! This hexadecimal notation, commonly referred to as [IPv6](https://en.wikipedia.org/wiki/IPv6), was developed in 2012 after we began running out of simpler IP addresses! 
+
+This address is known as an IP address, and this is what our computer systems still use. When you go to a *domain*, it looks up the underlying IP address, using the *Domain Name System* (DNS), and directs you to there. For example, Google's IP address is `216.58.219.46`. It's important to note that IP addresses are unique; no two servers can share the same IP address! Can you see why this may be a problem?
+
+> Check it out! Open Terminal if you're on a Mac ([instructions][ins-m], or Command Prompt if you're on a PC ([instructions][ins-w]. Type in `ping google.com` and see Google's IP address for yourself! 
+
+##### Domain Name System 
+
+Can you imagine if you had to memorize an IP address for every website you wanted to go to? That'd be incredibly troublesome; facebook wouldn't be `facebook.com`, it'd be `19.27.5.20`. It's obvious that the human mind isn't able to memorize arbitrary sets of numbers as well as it is able to remember meaningful names. To fill this need, the Domain Name System (DNS) was invented.
+
+The Domain Name System functions as almost as a digital equivlant of a yellow telephone book; it provides the functionality to lookup a *domain name* and get an equivalent *numerical IP address*. Since DNS is a distributed data store, no one server can contain the entire database; this is for safety as well as security: can you imagine if *the DNS* went down, or someone was able to hack into it? 
+
+Therefore, many different servers hosts a small portion of the database, and they all use a common set of software to communicate between each other. We call these computers *DNS servers*. Since all of these DNS servers can communicate between one another, if a DNS server does not contain the domain name requested, it simply redirects it to the next server which may contain the desired information. This way, you're continually redirected down the hierachy until you get to the domain name you request (as in the figure below). Neat, huh?
+
+![Image](https://web.stanford.edu/class/msande91si/www-spr04/readings/week1/InternetWhitepaper_files/ruswp_diag6.gif)
+
+Diagram sourced from [stanford.edu](stanford.edu).
```

Comment:
This is a broken link -- needs to be fully qualified with http:// etc

Reply:
Also can you source the diagram more specifically than just stanford.edu?